### PR TITLE
fix(ui): improve threshold settings validation

### PR DESCRIPTION
**Pull request by chnn**
This PR replaces the component we use for setting thresholds in gauge, single stat, and table views. 

threshold component

See #14190 for more context.

### CODE REVIEW

File: ui/src/shared/utils/thresholds.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  const thresholdType =
+    firstThresholdType === COLOR_TYPE_MIN ||
+    firstThresholdType === COLOR_TYPE_MAX
+      ? COLOR_TYPE_THRESHOLD
+      : firstThresholdType
+
+  const threshold = {
+    ...colorChoice,
+    id: uuid.v4(),
+    type: thresholdType, // TODO: should be same type as before?
```
Comment (chnn) [inline ui/src/shared/utils/thresholds.ts]:
```suggestion
    type: thresholdType,
```

File: ui/src/shared/components/ThresholdsSettings.test.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      {id: '0', type: 'threshold', name: 'fire', hex: '', value: 30},
+    ]
+
+    const {container} = render(
+      <ThresholdsSettings thresholds={thresholds} onSetThresholds={jest.fn()} />
+    )
+
+    // Enter an invalid value in the input
+    fireEvent.change(getInput(container, '0'), {
+      target: {value: 'baloney'},
```
Comment (OfTheDelmer) [inline ui/src/shared/components/ThresholdsSettings.test.tsx]: nice!

File: ui/src/shared/components/ThresholdsSettings.test.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    fireEvent.change(getInput(container, '1'), {
+      target: {value: '10'},
+    })
+
+    // Blur the input
+    fireEvent.blur(getInput(container, '1'))
+
+    // Expect an error message to exist
+    expect(getErrorMessage(container, '1')).toEqual(
+      'Please enter a value greater than the minimum threshold'
```
Comment (OfTheDelmer) [inline ui/src/shared/components/ThresholdsSettings.test.tsx]: great!

File: ui/src/shared/components/ThresholdsSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  return (
+    <div className="thresholds-settings">
+      <Button
+        className="thresholds-settings--add"
+        shape={ButtonShape.StretchToFit}
+        icon={IconFont.Plus}
+        text="Add a Threshold"
+        onClick={() => dispatch({type: 'THRESHOLD_ADDED'})}
+      />
+      {state.thresholds.map(threshold => {
```
Comment (OfTheDelmer) [inline ui/src/shared/components/ThresholdsSettings.tsx]: 🚀

**Pull request merged**